app/features/syllabus_generator/core.py

```diff
@@ -0,0 +1,30 @@
+from app.services.tool_registry import ToolFile
+from app.services.logger import setup_logger
+from app.features.syllabus_generator.tools import SyllabusBuilder
+from app.api.error_utilities import LoaderError, ToolExecutorError
+
+logger = setup_logger()
+
+def executor(grade_level: str, course_title: str, course_description: str, objectives_topics: str = None, required_materials: str = None, num_weeks: str = None, course_outline: str = None, grading_policy: str = None, class_policy: str = None, customization: str = None, verbose=False):
+    attributes = {}
+    attributes['grade_level'] = grade_level
```

Comment:
You can use the Pydantic schema for creating your own dict in base of their values: https://docs.pydantic.dev/1.10/usage/exporting_models/